### DISCUSSION TITLE
Wiki pages

### DISCUSSION THREAD

**Discussion by luskaner** [luskaner/ageLANServer#41] [Announcements]
Look at [QA](https://github.com/luskaner/aoe2DELanServer/wiki/Questions-and-Answers-(QA)) and [Troubleshooting](https://github.com/luskaner/aoe2DELanServer/wiki/Troubleshooting) pages if you have any doubts before creating an issue/discussion!

**Comment by jack3308** [comment]
### Context
I'm looking to have the server running in the background on a pi I have hosting a few other things for the house (DNS, DHCP, VPN, etc...). 

Most of the documentation seems to revolve around running the server on the same device you're playing on, but I would love to just leave the server running so my wife can potentially play with a friend if she wants (assuming they can copy the pre-configured launcher from a thumb-drive/network share). 

#### My question is this 
Can I just leave it running 24/7? Would you recommend I add some handling for restarts or crashes with a systemd service on the pi? Or can I just have the pi run it at startup and leave it at that? Just not quite sure how much of that stuff is already happening under the hood.

**Reply by luskaner** [reply]
Hello:

You certainly can leave it 24/7 as it automatically recovers any 'panics' it might encounter so it should not crash, however, since it has not been tested thoroughly for continuous operation, the more likely issue would be having a corrupted state which could impact one or more users (as everything is stored in-memory) particularly, related to creating or joining lobbies - or simply - panic continously. If you choose this path, I recommend having a way to restart the service, ideally at the start of every 'game session' or just once a day would be enough.

**Comment by Hulff** [comment]
hi im having this error 
Game age1.
Cleaning up...
Reverting all configuration and stopping its agent, authorize 'config-admin' if needed...
'Server' is already running, If you did not start it manually, kill the 'server' process using the task manager and execute the 'launcher' again.
Setting up...
Waiting 15 seconds for 'server' announcements on LAN on port(s) 31978 (we are v. 1), you might need to allow 'launcher' in the firewall...
Found a single 'server' "127.0.0.1, 192.168.0.105, <edited> (kubernetes.docker.internal.) - v. 1", will connect to it...
Adding host to hosts file, authorize 'config-admin-agent' if needed...
Failed to add host.
Exit code: 23.
Program finished with errors...
Pressione qualquer tecla para continuar. . .

**Reply by luskaner** [reply]
Ok, let's continue in the issue you created

**Comment by yeyuzh** [comment]
When I start the game and create an online game, it prompts that there is an advertisementJoinEvent error and I cannot create a multiplayer game.

**Reply by luskaner** [reply]
Hi:
Make sure when creating a lobby to select "Use Local LAN Server" in the Server dropdown. If that's not the problem please create a GitHub issue with your Server log for investigation.

**Comment by DuythuanBK** [comment]
Does this support Age of Empires (AoE1)?

**Reply by luskaner** [reply]
Hello:

Yeah it does support AoE I but only the Definitive Edition ;)

**Reply by DuythuanBK** [reply]
unfortunately, people in my country only play AOE original.

**Reply by luskaner** [reply]
Oh, I understand but the original Game has LAN without internet right? There is no need for a tool like mine I think

**Comment by MatinMadani** [comment]
Hi . I want to play aoe4 lan but the problem is that i dont have any access to internet via my computer but lan is available for play and the painful part is i did what you said exactly on both repos but the battle server manager shuts down itself immediately and even when i fully configured what you said it does not run at all and exits with exit code 20 after BattleServer.exe added to it fro aoe2 (before that error was 17). Please tell me what should i do to fix it i even tried the fully 100% offline solution but still the same issue.(I am on windows 11)

**Reply by luskaner** [reply]
Hello:

Currently, playing fully offline is very hard/not possible (I have not been able to do it as of now) as the game has more protections than the other games and my instructions on the Steam EMU don't work as-is, sorry :(

**Reply by MatinMadani** [reply]
Thank you for fast response. So what do you think i should do to make it able to be playing lan ? I managed to connect to internet but still the same error code 20 after trying to launch battle server manager it immediately shutdown , what is the problem?Would you tell me exactly ehat should i do to resolve the problem and play ?

**Reply by luskaner** [reply]
Hello:

Please in the launcher config.toml set `Log` to `true` re-run the launcher (let it do everything don't start the server first) and send me the logs (just make a zip) to luskaner at gmail dot com ;)

**Reply by MatinMadani** [reply]
Just sent to your email i hope it can help

**Reply by MatinMadani** [reply]
Just sent to your email i hope it can help

**Reply by luskaner** [reply]
@MatinMadani you battle-server-manager logs show lots of errors when trying to connect to the BattleServer.exe I'm pretty sure you are using an unsupported "BattleServer.exe", either it's not from Age of Empires II: Definitive Edition or is too old to work properly ;) 
Edit: or maybe is an issue with the firewall

**Reply by MatinMadani** [reply]
@luskaner thank you. I can ensure that firwall is not the issue. Would you mind if i ask you to send me you BattleServer.exe file in a zip file from gmail i sent you? I have very limited access to internet and if you do this it would be so much to me. I really appreciate your work and behavior.

**Reply by luskaner** [reply]
@MatinMadani sent you that, have you been able to make it work?

**Reply by MatinMadani** [reply]
@luskaner haven't received anything yet. Would you check it again if it is sent or not or even resend it again please?

**Reply by MatinMadani** [reply]
The error changed completely and i sent you the new logs. Thanks for your corporation.

**Comment by chenyr2022-blip** [comment]
Age of Empires IV I have successfully connected to the server and entered the game, but it shows:
We cannot verify you via Steam servers. You may continue playing in offline mode, but you will not earn XP.
Error Code:
CO0T01R-2X-0152656C69016112:00
Reconnect

**Reply by luskaner** [reply]
Hello:

Make sure your steam emulator is not configured in offline mode. Edit `steam_emu.ini` and change `Offline` to `0`

**Reply by chenyr2022-blip** [reply]
Thank you for your answers. The document has been settled successfully.

**Comment by PF3E06CN** [comment]
Hello,

First of all, thank you for your work on ageLANServer.

With the recent native release of Age of Empires II: Definitive Edition for macOS (Apple Silicon), I was wondering if there are any plans to support macOS clients in the future.

After inspecting the game binaries, it appears that the macOS version still uses Xbox/PlayFab components (XAL, XSAPI, PlayFab Party, PlayFab Multiplayer, etc.), which suggests that the networking stack may be quite similar to the Windows version.

Do you think macOS support would be feasible, or are there major technical obstacles that would prevent ageLANServer from working with the native Mac release?

I'd be happy to help test on Apple Silicon Macs if needed.

Thank you!

**Reply by luskaner** [reply]
Hello:

Thanks for the nice words :D
In this release I'm working on adding macOS supports, initially using CrossOver and Wine (I was working on it even before the native game was even announced) and intend to support the native version as well as stated in #531 . However, I'm working with an old mac os version (v13) on a Virtual Machine and I cannot really test much without GPU acceleration (and no arm64 so I cannot even test the native port) :( 

I was waiting for someone like you as I really need your help :D 
Please answer my questions in #531 and we will continue there ;)
